services/QuillLMS/client/app/bundles/Shared/components/fillInBlank/fillInBlankForm.tsx

```diff
@@ -0,0 +1,340 @@
+import React, { Component } from 'react';
+
+import {
+  ConceptSelector,
+  FlagDropdown,
+  TextEditor,
+} from '../../index';
+
+// Default instructions that appear when user types '/'
+const DEFAULT_FILL_IN_BLANKS_INSTRUCTIONS = [
+  'Fill in the blank with the correct joining word.',
+  'Fill in the blank with the correct word.',
+  'Fill in the blank with the correct pronoun.',
+  'Fill in the blank with the correct action word.',
+  'Fill in the blank with the correct set of words.',
+  'Fill in the blank with the correct option.'
+];
+
+interface FillInBlankFormState {
+  prompt: string;
+  blankAllowed: boolean;
+  caseInsensitive: boolean;
+  instructions: string;
+  cues: string;
+  conceptID: string;
+  flag: string;
+  cuesLabel: string;
+  newQuestionOptimalResponse: string;
+  showDefaultInstructions: boolean;
+}
+
+interface FillInBlankFormProps {
+  action: (data: any, optimalResponse?: any) => void;
+  editing?: boolean;
+  new?: boolean;
+  state?: Partial<FillInBlankFormState>;
+}
+
+export class FillInBlankForm extends Component<FillInBlankFormProps, FillInBlankFormState> {
+  constructor(props: FillInBlankFormProps) {
+    super(props);
+
+    // Initialize with default values, then merge in any provided state
+    const defaultState = {
+      prompt: '',
+      blankAllowed: false,
+      caseInsensitive: false,
+      instructions: '',
+      cues: '',
+      conceptID: '',
+      flag: 'alpha',
+      cuesLabel: '',
+      newQuestionOptimalResponse: '',
+      showDefaultInstructions: false
+    };
+
+    this.state = {
+      ...defaultState,
+      ...(props.state || {})
+    };
+  }
+
+  clearForm = () => {
+    this.setState({
+      blankAllowed: false,
+      caseInsensitive: false,
+      prompt: '',
+      newQuestionOptimalResponse: '',
+      instructions: '',
+      conceptID: '',
+      flag: 'alpha',
+      cuesLabel: '',
+      cues: '',
+      showDefaultInstructions: false
```

Comment:
It looks to me like this might be exactly the same values as those that are set above in `defaultState`.  If that's true, should we split this off into a shared constant that is referenced both there and here?